analyzer/tests/unit/test_checker_handling.py

```diff
@@ -94,14 +119,131 @@ def test_no_disabled_checks(self):
 
+    def test_checker_initializer(self):
+        """
+        Test initialize_checkers() function.
+        """
+        def add_description(checker):
+            return checker, ''
+
+        def all_with_status(status):
+            def f(checks, checkers):
+                result = set(check for check, data in checks.items()
+                             if data[0] == status)
+                return set(checkers) <= result
+            return f
+
+        args = []
+        context = MockContextSA()
+
+        # "security" profile, but alpha -> not in default.
+        security_profile_alpha = [
+                'alpha.security.ArrayBound',
+                'alpha.security.MallocOverflow']
+
+        # "default" profile.
+        default_profile = [
+                'security.FloatLoopCounter',
+                'deadcode.DeadStores']
+
+        # Checkers covering some "sei-cert" rules.
+        cert_guideline = [
+                'alpha.core.CastSize',
+                'alpha.core.CastToStruct']
+
+        checkers = []
+        checkers.extend(map(add_description, security_profile_alpha))
+        checkers.extend(map(add_description, default_profile))
+        checkers.extend(map(add_description, cert_guideline))
+
+        # "default" profile checkers are enabled explicitly. Others are in
+        # "default" state.
+        cfg_handler = ClangSA.construct_config_handler(args, context)
+        cfg_handler.initialize_checkers(context, checkers)
+        self.assertTrue(all_with_status(CheckerState.enabled)
+                        (cfg_handler.checks(), default_profile))
+        self.assertTrue(all_with_status(CheckerState.default)
+                        (cfg_handler.checks(), security_profile_alpha))
+
+        # "--enable-all" leaves alpha checkers in "default" state. Others
+        # become enabled.
+        cfg_handler = ClangSA.construct_config_handler(args, context)
+        cfg_handler.initialize_checkers(context, checkers, enable_all=True)
+        self.assertTrue(all_with_status(CheckerState.default)
+                        (cfg_handler.checks(), security_profile_alpha))
+        self.assertTrue(all_with_status(CheckerState.enabled)
+                        (cfg_handler.checks(), default_profile))
+
+        # Enable alpha checkers explicitly.
+        cfg_handler = ClangSA.construct_config_handler(args, context)
+        cfg_handler.initialize_checkers(context, checkers, [('alpha', True)])
+        self.assertTrue(all_with_status(CheckerState.enabled)
+                        (cfg_handler.checks(), security_profile_alpha))
+        self.assertTrue(all_with_status(CheckerState.enabled)
+                        (cfg_handler.checks(), default_profile))
+
+        # Enable "security" profile checkers.
+        cfg_handler = ClangSA.construct_config_handler(args, context)
+        cfg_handler.initialize_checkers(context, checkers,
+                                        [('profile:security', True)])
```

Comment:
I tried to run an analysis with the following command `CodeChecker analyze -o ./reports ./build.json -e profile:security` but the command line gives me the following warning message:
```
[WARNING 2020-07-31 09:50] - No checker(s) with these names was found:
profile:security
```
It will give me the same warning message when the profile actually doesn't exist: `CodeChecker analyze -o ./reports ./build.json -e dummy`

If the profile exists I don't want to see this warning message but if a profile doesn't exist I would like to see a warning message that the given profile doesn't exist. Same for guidelines.